profile.md

```diff
@@ -486,10 +486,13 @@ Examples values of the `scope` claim:
 
 ### Group Based Authorization: wlcg.groups 
 
-Authorization may be based on the `wlcg.groups` claim. The value of the `wlcg.groups` claim is an ordered JSON array of case-sensitive strings reflecting the VO groups of which the token subject is a member. 
+Authorization may be based on the `wlcg.groups` claim. The value of the `wlcg.groups` claim is an ordered JSON array of case-sensitive strings reflecting the assertion, by the issuer, of group membership of the subject of the token.
 
-`wlcg.groups `semantics are equivalent to existing VOMS groups. VOMS roles should be considered as optional (i.e. returned only if requested) `wlcg.groups`; selection of optional groups is discussed below.
+For authorization decisions, relying parties MUST NOT consider any group memberships that are **not** asserted by the OP in the token.  Particularly, membership in a child group (e.g. `/cms/uscms`) does not imply membership in the parent group (`/cms`).
```

Comment:
Hi Paul,
while in practice it may remain unlikely for a member of a subgroup not to be a member also of its parent group(s), there could be a use case for that.  For example, a robot entity might be made a member just of a subgroup created for activities pertaining to that robot entity; if a wider membership is not really needed, it would seem best not to empower that entity with it.

That said, the actual matter is the following: if a group that is presented in the token implies membership of all the corresponding parent groups, each presented group would have to be _parsed_ and we would need to define a language for expressing group notions.  AFAIR, this has been discussed in AuthZ WG meetings and the majority consensus at the time was that we prefer groups to be treated as _opaque strings_ for which simple string comparisons suffice to determine which group privileges to bestow on the client presenting the token.

Reply:
While the document does spell out (in the Common Claims section) the _grammar_ for expressing the group memberships, it currently says nothing about the _semantics_ of groups.

Reply:
FWIW, I agree with the decision.  I think it reasonable for an RP to consider groups as opaque tokens.

I'm not particularly keen of AARC G069's requirements, as RPs need to parse the group-membership assertions and (potentially) build the implied parent hierarchy.

My points where intended to point out that, by supporting people who are members of subgroups but not a member of the parent group, JWT profile is directly conflicting the AARC group membership model.

In the longer term, we want to harmonise these two approaches.  Therefore, (I think) we should be very sure we need to support this use-case.  We currently don't have it with VOMS, so why do we need it now?

Reply:
Currently, in IAM, the members of a subgroup are also members of the parent group and a token always includes all groups to which a user belongs, so there is no need to second-guess.
Personally, I find the current hierarchical model the easiest to reason about and, since it has served us well in the past decades, I wouldn't abandon it lightly.

Reply:
I have no strong preference. Just some points to keep in mind:
- inherited membership can reduce the number of groups to be asserted hence reduce the size of a token
- changing current behaviour is hard
- inherited membership is probably more prone to configuration errors
- inherited membership is more difficult to parse.